polls/utils.py

```diff
@@ -32,10 +33,49 @@ def check_user_work_permission(user):
         and time_diff_minutes < minutes_after_can_continue
     )
     can_continue = time_diff_minutes > minutes_after_can_continue
-    rest_time = round(minutes_after_can_continue - time_diff_minutes) + 1
+    rest_time = round(minutes_after_can_continue - time_diff_minutes)
     return can_continue, should_rest, rest_time
 
 
+def get_user_num_tasks(user):
+    """Generate number of task a user completed then return
+    the value to be used in template tags
+    """
+    task_count = Annotation.objects.get(user=user).annotate(total_task=Count("task"))
+
+    return task_count["total_task"]
+
+
+def get_num_user_gold_task(user):
+    """Generate the number of gold tasks a user completed then return
+    it's value for use in template tags
+    """
+    gold_count = (
+        Annotation.objects.get(user=user)
```

Comment:
Should be using filter in place of gte

Reply:
If going to filter then should assert the list is length one. But I think this is nitpicking, we can have a global contract that user is always just one

Reply:
> If going to filter then should assert the list is length one. But I think this is nitpicking, we can have a global contract that user is always just one

I guess we can simply filter annotations for a particular user with gold task and get count. Wouldn't that be better?

Reply:
@najamF18 my mistake, I thought this was retrieving the user.

Reply:
> 

No worries. Got it